lua/neo-tree/sources/common/components.lua

```diff
@@ -15,9 +15,54 @@ local utils = require("neo-tree.utils")
 local file_nesting = require("neo-tree.sources.common.file-nesting")
 local container = require("neo-tree.sources.common.container")
 local log = require("neo-tree.log")
+local path = require("plenary.path")
 
 local M = {}
 
+-- Workaround for https://github.com/nvim-lua/plenary.nvim/issues/411
+local relative_path = function(original_path, reference_path)
+  -- Use plenary's make_relative to clean paths
+  original_path = path:new(original_path):make_relative(".")
+  reference_path = path:new(reference_path):make_relative(".")
+  local o_path = path:new(original_path)
+  local ref_path = path:new(reference_path)
+  local parents = o_path:parents()
+  local ref_parents = ref_path:parents()
+
+  local path_elements = vim.split(o_path.filename, "/")
+  table.insert(parents, 1, original_path)
+  table.insert(ref_parents, 1, reference_path)
+
+  local result = ""
+  for i, ref_parent in ipairs(ref_parents) do
+    for j, par in ipairs(parents) do
+      if ref_parent == par then
+        if i == 1 and j == 1 then
+          return ""
+        end
+
+        result = result .. table.concat(path_elements, "/", #path_elements - j + 2)
+        return result
+      end
+    end
+
+    result = "../" .. result
+  end
```

Comment:
This might be a simpler (and technically faster) algorithm, although I haven't tested it much:
```suggestion
    if o_path.path.root ~= ref_path.path.root then
      return o_path:absolute()
    end
    local i, j = #parents, #ref_parents
    -- find the deepest common directory
    while parents[i] == ref_parents[j] and math.min(j, i) > 0 do
      i = i - 1
      j = j - 1
    end
    local common_dir = parents[i + 1]
    local steps_out = string.rep(".." .. sep, j)
    return steps_out .. original_path:sub(#common_dir + 2)
  end
```